### DISCUSSION TITLE
Grafana For Charting

### DISCUSSION THREAD

**Discussion by five2seven** [babybuddy/babybuddy#497] [Q&A]
I have Grafana in Docker already for the data from a speed test container and I'd love to be able to feed BabyBuddy data into it as well. There are dashboards on the Grafana website for Django apps but they also need a Prometheus instance and it's all a bit over my head. Has anyone pulled their BB data into a Grafana dashboard?

**Comment by ahgray** [comment]
Probably not what  you're looking for, but. Babybuddy already has a Home Assistant integration, and once you have your baby buddy data in Home Assistant, it integrates directly with Grafana, plus lots of other functionality. Like i said not a direct answer to your quest but it is a pretty much turn-key option at this point with no development.

**Reply by five2seven** [reply]
This isn't a bad idea, I just don't know how to implement. I installed the Grafana add-on but don't see how to configure getting the BB data into it. Is this something you have done? If so, can you share what your sanitized configs are?

**Comment by ahgray** [comment]
Sounds like your half way there. Without getting into the nitty gritty the top level architecture should be something like this.

Grafana Add-on For Home Assistant <-------> Home Assistant <--------> Baby Buddy Home Assistant Integration(IN HACS) <-----------> Baby Buddy

So its 4 Parts that work together, and the way I have it implemented, I have Home Assistant on a VM locally, and Baby Buddy on Heroku(because i was too lazy to set up port forwarding/ proxy etc.) So its only got two separate pieces, There is a Baby Buddy home assistant add-on as well so theoretically you could have all 4 in one self-contained vm.

As for the Graffana part anything that you have tied into Home Assistant becomes available for Grafana to query, I've got everything in there from my plant monitors, to my BBQ Probes.
